src/main/java/dataStructures/binarySearchTree/README.md

```diff
@@ -0,0 +1,67 @@
+# Binary Search Tree
```

Comment:
@junnengsoo great. Under the Notes section, can mention this property is often the base for several balanced tree structures, like AVL